liquidctl/driver/commander_core.py

```diff
@@ -241,8 +265,10 @@ def _wake_device_context(self):
         finally:
             self._send_command(_CMD_SLEEP)
 
-    def _write_data(self, mode, data_type, data):
-        self._read_data(mode, data_type)  # Will ensure we are writing the correct data type to avoid breakage
+    def _write_data(self, mode, data_type, data, force=False):
+        if not force:
+            # Ensure we are writing the correct data type to avoid breakage
+            self._read_data(mode, data_type)
```

Comment:
I'm not so sure about this anymore: can't this end up causing the very issue we're trying to autocorrect, because of the suppression of this "write-the-correct-data-type" read during the second write (`_DATA_TYPE_HW_FIXED_PERCENT`)?

Should `force=True` only suppress the comparison inside `_read_data`, instead of the entire call to it?

Reply:
I think this is fine.